From foo@baz Wed Jun  3 10:17:47 PDT 2009
Date: Wed, 03 Jun 2009 10:17:47 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: pata_rdc: remove broken flag

The "in_module_init" flag was wrong, so just remove it, it's not needed.


Cc: Kevin Huang  <Kevin.Huang@rdc.com.tw>
Cc: Tomy Wang <Tomy.Wang@rdc.com.tw>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/pata_rdc/pata_rdc.c |   11 -----------
 1 file changed, 11 deletions(-)

--- a/drivers/staging/pata_rdc/pata_rdc.c
+++ b/drivers/staging/pata_rdc/pata_rdc.c
@@ -38,10 +38,6 @@ static const struct pci_device_id rdc_pa
 };
 MODULE_DEVICE_TABLE(pci, rdc_pata_id_table);
 
-static unsigned int in_module_init = 1; /* hotplugging check??? */
-
-/* ata device data */
-
 /* see ATA Host Adapters Standards. */
 static struct pci_bits ATA_Decode_Enable_Bits[] = {
 	{ 0x41U, 1U, 0x80UL, 0x80UL },	/* port (Channel) 0 */
@@ -982,11 +978,6 @@ static int __devinit rdc_init_one(struct
 
 	dbgprintf("rdc_init_one\n");
 
-	/* no hotplugging support (FIXME) */ /* why??? */
-	if (!in_module_init) {
-		dbgprintf("rdc_init_one in_module_init == 0 failed \n");
-		return -ENODEV;
-	}
 	port_info[0] = rdc_pata_port_info[ent->driver_data];
 	port_info[1] = rdc_pata_port_info[ent->driver_data];
 
@@ -1026,8 +1017,6 @@ static int __init pata_rdc_init(void)
 		return rc;
 	}
 
-	in_module_init = 0;
-
 	return 0;
 }
 
